pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -330,6 +330,26 @@ else let
         ++ optional (elem "host"   configurePlatforms) "--host=${stdenv.hostPlatform.config}"
         ++ optional (elem "target" configurePlatforms) "--target=${stdenv.targetPlatform.config}";
 
+      cmakeFlags = let inherit (lib) optionals; in
+        (/**/ if lib.isString cmakeFlags then lib.warn "String 'cmakeFlags' is deprecated and will be removed in release 23.05. Please use a list of strings. Derivation name: ${derivationArg.name}, file: ${pos.file or "unknown file"}" [cmakeFlags]
+         else if cmakeFlags == null      then lib.warn "Null 'cmakeFlags' is deprecated and will be removed in release 23.05. Please use a empty list instead '[]'. Derivation name: ${derivationArg.name}, file: ${pos.file or "unknown file"}" []
+         else                                     cmakeFlags)
+        ++ optionals (stdenv.hostPlatform != stdenv.buildPlatform) (
+          [ "-DCMAKE_SYSTEM_NAME=${lib.findFirst lib.isString "Generic" (lib.optional (!stdenv.hostPlatform.isRedox) stdenv.hostPlatform.uname.system)}"]
+        ++ optionals (stdenv.hostPlatform.uname.processor != null)
+          [ "-DCMAKE_SYSTEM_PROCESSOR=${stdenv.hostPlatform.uname.processor}" ]
+        ++ optionals (stdenv.hostPlatform.uname.release != null)
+          [ "-DCMAKE_SYSTEM_VERSION=${stdenv.hostPlatform.uname.release}" ]
+        ++ optionals (stdenv.hostPlatform.isDarwin)
+          [ "-DCMAKE_OSX_ARCHITECTURES=${stdenv.hostPlatform.darwinArch}" ]
+        ++ optionals (stdenv.buildPlatform.uname.system != null)
+          [ "-DCMAKE_HOST_SYSTEM_NAME=${stdenv.buildPlatform.uname.system}" ]
+        ++ optionals (stdenv.buildPlatform.uname.processor != null)
+          [ "-DCMAKE_HOST_SYSTEM_PROCESSOR=${stdenv.buildPlatform.uname.processor}" ]
+        ++ optionals (stdenv.buildPlatform.uname.release != null)
+          [ "-DCMAKE_HOST_SYSTEM_VERSION=${stdenv.buildPlatform.uname.release}" ]
```

Comment:
I actually liked the more compact form it had before more

Reply:
switched it to be more compact